src/coreclr/vm/jitinterface.cpp

```diff
@@ -11190,8 +11190,46 @@ void CEEJitInfo::BackoutJitData(EEJitManager * jitMgr)
     CodeHeader* pCodeHeader = m_CodeHeader;
     if (pCodeHeader)
         jitMgr->RemoveJitData(pCodeHeader, m_GCinfo_len, m_EHinfo_len);
+
+    delete [] (BYTE*)m_CodeHeaderRW;
+    m_CodeHeaderRW = NULL;
 }
 
+/*********************************************************************/
+void CEEJitInfo::WriteCode(EEJitManager * jitMgr)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+    } CONTRACTL_END;
+
+#ifdef USE_INDIRECT_CODEHEADER
+    if (m_pRealCodeHeader != NULL)
+    {
+        // Restore the read only version of the real code header
+        m_CodeHeaderRW->SetRealCodeHeader(m_pRealCodeHeader);
+        m_pRealCodeHeader = NULL;
+    }
+#endif // USE_INDIRECT_CODEHEADER
+
+    memcpy(m_CodeHeader, m_CodeHeaderRW, m_codeWriteBufferSize);
+    delete [] (BYTE*)m_CodeHeaderRW;
+    m_CodeHeaderRW = NULL;
+    m_codeWriteBufferSize = 0;
+
```

Comment:
```suggestion
```
This can be taken care of by the destructor

Reply:
Done